awx/api/serializers.py

```diff
@@ -3382,10 +3382,16 @@ class Meta:
 class WorkflowApprovalSerializer(UnifiedJobSerializer):
 
     can_approve_or_deny = serializers.SerializerMethodField()
+    approval_expiration = serializers.SerializerMethodField()
 
     class Meta:
         model = WorkflowApproval
-        fields = (['*', '-controller_node', '-execution_node', 'can_approve_or_deny'])
+        fields = (['*', '-controller_node', '-execution_node', 'can_approve_or_deny', 'approval_expiration'])
+
+    def get_approval_expiration(self, obj):
+        if obj.status != 'pending' or obj.timeout == 0:
+            return None
+        return now() + timedelta(seconds=obj.timeout)
```

Comment:
If this is the date/time the approval *will* expire, this should probably be more like `obj.created + timedelta(seconds=obj.timeout)`.

Reply:
Oh you're right. When we were spitballing this, my brain had not gotten over the choice between "will expire on Friday" versus "will expire in 2 days". The current ask from @mabashian is the Friday version, which is what @ryanpetrello's suggested implementation will do.